airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -276,12 +428,42 @@ public void writeDestinationOAuthParam(final DestinationOAuthParameter destinati
     return persistence.listConfigs(ConfigSchema.DESTINATION_OAUTH_PARAM, DestinationOAuthParameter.class);
   }
 
+  // todo: should replaceAllConfigs just write individual configs using their proper call instead of
```

Comment:
would love some input on this

Reply:
This is related to conversation about what it means to export a dump of configuration, and re-import it later, as it pertains to secrets.  I think secrets should be a separate dump, and the "usual" backup of airbyte config should be the non-secret version of the connections.  That way it can more safely be used to copy a prod config to a dev server for teams that need that.  This may need more discussion about user goals.

As for breaking out the code... I think there is implementation awkwardness no matter which way we do it, because if it goes in the persistence layer, some persistences will have "not implemented", and if it goes in the repository layer, the repository starts having responsibility for details it otherwise ignores.  I don't think there's a perfect solution.